lib/Pipes/TaggedFunctionKind.cpp

```diff
@@ -50,6 +52,107 @@ using TaggedFK = TaggedFunctionKind;
 void TaggedFK::getInvalidations(const Context &Ctx,
                                 TargetsList &ToRemove,
                                 const GlobalTupleTreeDiff &Diff) const {
+  return;
+}
+
+using IsolatedFK = IsolatedFunctionKind;
+void IsolatedFK::getInvalidations(const Context &Ctx,
+                                  TargetsList &ToRemove,
+                                  const GlobalTupleTreeDiff &Diff) const {
+  using PM = PathMatcher;
+  using Binary = model::Binary;
+
+  const auto &CurrentModel = getModelFromContext(Ctx);
+  if (not Ctx.containsReadOnlyContainer(BinaryCrossRelationsRole))
+    return;
+
+  static constexpr auto BCRR = BinaryCrossRelationsRole;
+  const auto &Container = Ctx.getReadOnlyContainer<FileContainer>(BCRR);
+
+  auto MaybeContainerPath = Container.path();
+  revng_assert(MaybeContainerPath.has_value());
+  auto MaybeBuffer = llvm::MemoryBuffer::getFile(*MaybeContainerPath);
+  revng_assert(MaybeBuffer);
+  yaml::Input YAMLInput((*MaybeBuffer)->getBuffer());
+
+  yield::CrossRelations Relations;
+  YAMLInput >> Relations;
+
+  const auto *ModelDiff = Diff.getAs<Binary>();
+  if (not ModelDiff)
+    return;
```

Comment:
@drblallo Doesn't the diff include which global is subject to the change? We can't decide how to at only based on type.

Reply:
sent to antonio the fix